HW7/src/com/jvprogram/sinhronizer/barjer/Robot.java

```diff
@@ -0,0 +1,26 @@
+package com.jv_program.sinhronizer.barjer;
+
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+
+public class Robot extends Thread {
+    private int id;
+    private CyclicBarrier cyclBarrier;
+
+    public Robot (int id, CyclicBarrier cyclBarrier) {
+        this.id = id;
+        this.cyclBarrier = cyclBarrier;
+    }
+
+    @Override
+    public void run () {
+        try {
+            System.out.println("Robot: " + id + " finished his work.");
+            cyclBarrier.await();
+        } catch (InterruptedException e) {
```

Comment:
Since the processing of the exception is the same ( e.printStackTrace();) you could write
 catch (InterruptedException|BrokenBarrierException e)